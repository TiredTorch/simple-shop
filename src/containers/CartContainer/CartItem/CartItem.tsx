import Image from "next/image";
import { FC } from "react";
import { Button } from "@/components";
import { CartItemProps } from "./CartItem.types";

export const CartItem: FC<CartItemProps> = () => {
    return (
        <div className="flex items-center w-full justify-between bg-slate-500 px-8 py-4 text-slate-800 rounded-sm">
            <div className="flex gap-6">
                <Image src={""} alt="" />
                <div className="font-semibold">name</div>
                <div>amount</div> {/*TODO add counter to change amount*/}
                <div className="font-semibold">total price</div>
            </div>
            <div className="flex gap-6">
                <Button buttonVariation={"cartItemButton"}>Order</Button>
                <Button buttonVariation={"cartItemButton"}>Remove</Button>
            </div>
        </div>
    );
};
