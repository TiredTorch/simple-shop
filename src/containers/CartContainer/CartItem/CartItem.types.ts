import { CartItemDto } from "@/types";

export type CartItemProps = CartItemDto & {
    handleOrder: (amount: number) => () => void;
    handleRemove: VoidFunction;
};
