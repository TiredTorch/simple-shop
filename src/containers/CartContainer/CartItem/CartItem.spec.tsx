import { faker } from "@faker-js/faker";
import { render } from "@testing-library/react";
import { CartItem } from "./CartItem";

describe("CartItem", () => {
    it("should render cart item", () => {
        const handleOrderMock = jest.fn();
        const handleRemoveMock = jest.fn();

        const imageUrlMock = faker.image.url();
        const nameMock = faker.person.firstName();
        const amountMock = faker.number.int();
        const priceMock = faker.number.int();

        const { container } = render(
            <CartItem
                image={imageUrlMock}
                name={nameMock}
                amount={amountMock}
                pricePerOne={priceMock}
                handleOrder={handleOrderMock}
                handleRemove={handleRemoveMock}
            />
        );

        expect(container).toMatchSnapshot();
    });
});
