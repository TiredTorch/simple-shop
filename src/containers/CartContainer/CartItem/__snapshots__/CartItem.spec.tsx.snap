// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem should render cart item 1`] = `
<div>
  <div
    class="flex items-center w-full justify-between bg-slate-500 px-8 py-4 text-slate-800 rounded-sm"
  >
    <div
      class="flex gap-6 items-start xl:items-center flex-col xl:flex-row xl:gap-16"
    >
      <img
        alt="itemImage"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2FG1Far%2F640%2F480&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2FG1Far%2F640%2F480&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2FG1Far%2F640%2F480&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2FG1Far%2F640%2F480&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2FG1Far%2F640%2F480&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2FG1Far%2F640%2F480&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2FG1Far%2F640%2F480&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2FG1Far%2F640%2F480&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2FG1Far%2F640%2F480&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <div
        class="font-semibold"
      >
        Asha
      </div>
      <div
        class="flex items-center gap-1"
      >
        <button
          class="
                undefined
                rootButton
                cartAmountButton
            "
        >
          -
        </button>
        6119792004038656
        <button
          class="
                undefined
                rootButton
                cartAmountButton
            "
        >
          +
        </button>
      </div>
      <div
        class="font-semibold"
      >
        9.955689452858738e+30
        $
      </div>
    </div>
    <div
      class="flex gap-6 flex-col xl:flex-row"
    >
      <button
        class="
                undefined
                rootButton
                cartItemButton
            "
      >
        Order
      </button>
      <button
        class="
                undefined
                rootButton
                cartItemButton
            "
      >
        Remove
      </button>
    </div>
  </div>
</div>
`;
